src/analysis/retail/warlock/demonology/modules/talents/Doom.tsx

```diff
@@ -17,22 +24,79 @@ class Doom extends Analyzer {
     enemies: Enemies,
   };
   enemies!: Enemies;
-  damage = 0;
+  doom = {
+    applyDebuffCount: 0,
+    removeDebuffCount: 0,
+    damage: 0,
+    hits: 0,
+  };
+
+  demonboltCasts = 0;
+  demonboltWithCoreCasts = 0;
 
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS.DOOM_TALENT);
+
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.DOOM),
+      this.onDoomApply,
+    );
+    this.addEventListener(
+      Events.removedebuff.by(SELECTED_PLAYER).spell(SPELLS.DOOM),
+      this.onDoomRemove,
+    );
+    // Listen for Doom damage events (uses different spell ID than debuff)
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DOOM_DAMAGE),
+      this.onDoomDamage,
+    );
     this.addEventListener(
-      Events.damage.by(SELECTED_PLAYER).spell(TALENTS.DOOM_TALENT),
-      this.handleDoomDamage,
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEMONBOLT),
+      this.onDemonboltCast,
     );
   }
 
-  handleDoomDamage(event: DamageEvent) {
-    this.damage += event.amount + (event.absorbed || 0);
+  onDoomApply(event: ApplyDebuffEvent) {
+    this.doom.applyDebuffCount += 1;
+  }
+
+  onDoomRemove(event: RemoveDebuffEvent) {
+    this.doom.removeDebuffCount += 1;
+  }
+
+  onDoomDamage(event: DamageEvent) {
+    this.doom.hits += 1;
+    const damage = event.amount + (event.absorbed || 0);
+    this.doom.damage += damage;
+  }
+
+  onDemonboltCast(event: CastEvent) {
+    this.demonboltCasts += 1;
+
+    // Check if player has Demonic Core buff when casting Demonbolt
+    const hasCore = this.selectedCombatant.hasBuff(SPELLS.DEMONIC_CORE_BUFF.id, event.timestamp);
+    if (hasCore) {
+      this.demonboltWithCoreCasts += 1;
+    }
   }
+
   get uptime() {
-    return this.enemies.getBuffUptime(TALENTS.DOOM_TALENT.id) / this.owner.fightDuration;
+    return this.enemies.getBuffUptime(SPELLS.DOOM.id) / this.owner.fightDuration;
+  }
+
+  get doomHitsPerExpire() {
+    return this.doom.removeDebuffCount > 0 ? this.doom.hits / this.doom.removeDebuffCount : 0;
+  }
+
+  get doomDamagePerExpire() {
+    return this.doom.removeDebuffCount > 0 ? this.doom.damage / this.doom.removeDebuffCount : 0;
+  }
+
+  get doomApplicationRate() {
+    return this.demonboltWithCoreCasts > 0
+      ? this.doom.applyDebuffCount / this.demonboltWithCoreCasts
+      : 0;
```

Comment:
These don't appear to be used anywhere, is that intentional?

Reply:
Unintentional! removed!